lib/Modules/ComputeHooks/ComputeHooks.psm1

```diff
@@ -106,1360 +56,800 @@ function Install-Prerequisites {
+    $cfg = Get-JujuCharmConfig
+    $adGroup = "{0}\{1}" -f @($adCtxt['netbiosname'], $cfg['ad-computer-group'])
+    $adUser = $adCtxt['adcredentials'][0]["username"]
 
-    if ($net_type -eq "ovs"){
-        $neutron_restart = Generate-Config -ServiceName "neutron-ovs"
-    }else{
-        $neutron_restart = Generate-Config -ServiceName "neutron"
-    }
-
-    $nova_restart = Generate-Config -ServiceName "nova"
-    $JujuCharmServices = Get-CharmServices
-
-    if ($nova_restart){
-        Write-JujuInfo "Restarting service Nova"
-        Restart-Nova
-    }
+    $marshaledAccounts = Get-MarshaledObject -Object @($adGroup, $adUser)
 
-    if ($neutron_restart){
-        Write-JujuInfo "Restarting service Neutron"
-        Restart-Neutron
-    }
-    Open-CharmPorts
-    if($nova_restart -and $neutron_restart){
-        Set-JujuStatus -Status active -Message "Unit is ready"
+    $relationSettings = @{
+        'accounts' = $marshaledAccounts
     }
-}
 
-function Start-InstallHook {
-    if(!(Get-IsNanoServer)){
-        try {
-            Set-MpPreference -DisableRealtimeMonitoring $true
-        } catch {
-            # No need to error out the hook if this fails.
-            Write-JujuWarning "Failed to disable antivirus: $_"
-        }
-    }
-    # Set machine to use high performance settings.
-    try {
-        Set-PowerProfile -PowerProfile Performance
-    } catch {
-        # No need to error out the hook if this fails.
-        Write-JujuWarning "Failed to set power scheme."
-    }
-    Start-TimeResync
-    $netbiosName = Convert-JujuUnitNameToNetbios
-    $hostnameChanged = Get-CharmState -Namespace "Common" -Key "HostnameChanged"
-    $hostnameReboot = $false
-    if (!($hostnameChanged) -and ($COMPUTERNAME -ne $netbiosName)) {
-        Write-JujuWarning ("Changing computername from {0} to {1}" -f @($COMPUTERNAME, $netbiosName))
-        Rename-Computer -NewName $netbiosName
-        Set-CharmState -Namespace "Common" -Key "HostnameChanged" -Value $true
-        $hostnameReboot = $true
-    }
-    $prereqReboot = Install-Prerequisites
-    if ($hostnameReboot -or $prereqReboot) {
-        Invoke-JujuReboot -Now
+    $rids = Get-JujuRelationIds 'cinder-accounts'
+    foreach($rid in $rids) {
+        Set-JujuRelation -RelationId $rid -Settings $relationSettings
     }
-    Import-CloudbaseCert
-    Start-ConfigureVMSwitch
-    $installerPath = Get-NovaInstaller
-    Install-Nova -InstallerPath $installerPath
-    Confirm-ServicePrerequisites
-    Start-ConfigureNeutronAgent
-    Enable-MSiSCSI
 }
 
-function Start-ADJoinRelationChangedHook {
-    $adCtxt = Get-ActiveDirectoryContext
-    if(!$adCtxt["adcredentials"]) {
-        Write-JujuWarning "AD user credentials are not already set"
-        return
-    }
-    if (Get-IsNanoServer) {
-        # Due to a bug that throw CIM exception when setting services under AD
-        # user, we run the services as LocalSystem in Nano for the moment.
+function Start-LocalMonitorsRela
```

Comment:
what the? This was supposed to be here:

https://github.com/cloudbase/nova-hyperv-charm/pull/2/files#diff-382ecbd9c9cbd3bf96f0b9293e144264R815